src/host/sparse.jl

```diff
@@ -0,0 +1,79 @@
+## General Sparse Matrix
+
+Base.size(A::AbstractGPUSparseMatrix) = (A.m, A.n)
+
+SparseArrays.nonzeros(A::AbstractGPUSparseMatrix) = A.nzval
+SparseArrays.getnzval(A::AbstractGPUSparseMatrix) = nonzeros(A)
+SparseArrays.nnz(A::AbstractGPUSparseMatrix) = length(nzval(A))
+
+function LinearAlgebra.rmul!(A::AbstractGPUSparseMatrix, x::Number)
+    rmul!(SparseArrays.getnzval(A), x)
+    return A
+end
+
+function LinearAlgebra.lmul!(x::Number, A::AbstractGPUSparseMatrix)
+    lmul!(x, SparseArrays.getnzval(A))
+    return A
+end
+
+## CSC Matrix
+
+SparseArrays.getcolptr(A::AbstractGPUSparseMatrixCSC) = A.colptr
+SparseArrays.rowvals(A::AbstractGPUSparseMatrixCSC) = A.rowval
```

Comment:
I don't think this is a great interface. Why not defer to the backend-specific type to implement these methods?

Reply:
Do you mean to directly iimplement there in their specific backends? I was also thinking on that, but how can I perform the generic `mul!` case then? There I need to extract those data.

Reply:
> how can I perform the generic `mul!` case then

By calling `SparseArrays.getcolptr` (as implemented by the back-end) in the kernel. Which is also how the stdlib does it: https://github.com/JuliaSparse/SparseArrays.jl/blob/4fd3aad5735e3b80eefe7b068f3407d7dd0c0924/src/linalg.jl#L561-L562

Reply:
Ok, what I had in mind was to define those methods in general. And then each backend can apply its own methods. 

If you think better not to do this, we can remove it.

Reply:
You cannot define them in general, though. There's no guarantee that concrete types have fields named like that.

Reply:
Ok, then I will remove this definition.